dsl/pace/dsl/stencil.py

```diff
@@ -953,14 +1043,22 @@ def restrict_vertical(self, k_start=0, nk=None) -> "StencilFactory":
             grid_indexing=self.grid_indexing.restrict_vertical(k_start=k_start, nk=nk),
         )
 
+    def show_build_report(self, key: str = "build_time", **kwargs) -> str:
```

Comment:
nit: you could remove "show" from these method names (possibly replacing with "get"), since this doesn't actually show (print) a report, just returns it as a string.